tests/scripts/all.sh

```diff
@@ -189,6 +189,10 @@ pre_initialize_variables () {
     if [ -z "${MAKEFLAGS+set}" ]; then
         export MAKEFLAGS="-j$(all_sh_nproc)"
     fi
+    # if CC is not set, use clang by default (if present) to improve build times
+    if [ -z "${CC+set}" ] && (type clang > /dev/null 2>&1); then
+        export CC=$(type -p clang)
```

Comment:
Why not just `export CC=clang`? We've established from the line above that `clang` is a valid command (which includes executables, functions and aliases) but if it's not an actual file on disk then this will set `CC` to empty

Reply:
Originally I did just have `clang`, but changed to this because `scripts/mbedtls_dev/c_build_helper.py` calls `$CC` via the subprocess module, without `shell=True`, which I think means python won't search the path for it. This change didn't solve the problem I was having there, but I left the full path here anyway. No particular reason not to change it back.

Reply:
Nearly all the time I imagine people are going to set `CC=gcc`, `CC=clang`, `CC=gcc-10`, `CC=clang-13`, etc, i.e. without a path, so it seems reasonable that this script should do the same

Reply:
The Python `subprocess` function do path lookup on the command name even with `shell=False`. Passing the full path here only makes a difference if `PATH` changes, which is not something we currently do, but it's arguably more robust to select a specific clang version here (if a component or a called script wants a different clang, it will override `CC` anyway).

Aliases and functions are not visible to subprocesses no matter what, so if there is an alias or function called `clang` here, it should be discarded. But we don't define any such alias or function in this script, and doing that would be obviously disruptive so it isn't something we need to be robust against.